assert_eq! too strict re strings

The following test passes using the standard `assert_eq!`, but fails to compile using `pretty_assertions`:
```rust
    #[test]
    fn it_works() {
	let s0: &'static str = "foo";
	let s1: String = "foo".to_string();
	assert_eq!(s0, s1);
    }
```

The error is
```
error[E0308]: mismatched types
  --> src/lib.rs:10:2
   |
10 | 	assert_eq!(s0, s1);
   | 	^^^^^^^^^^^^^^^^^^^ expected &str, found struct `std::string::String`
   |
   = note: expected type `&&str`
              found type `&std::string::String`
   = note: this error originates in a macro outside of the current crate

```

Please can the behaviour of `assert_eq!` be relaxed, so `pretty_assertions` can truly be a drop-in replacement? Thanks.

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -76,7 +76,7 @@ pub struct Comparison {
 }
 
 impl Comparison {
-    pub fn new<T: Debug>(left: &T, right: &T) -> Comparison {
+    pub fn new<TLeft: Debug, TRight: Debug>(left: &TLeft, right: &TRight) -> Comparison {
         let left_dbg = format!("{:?}", *left);
         let right_dbg = format!("{:?}", *right);
         let changeset = Changeset::new(&left_dbg, &right_dbg, " ");

--- a/tests/assert_eq.rs
+++ b/tests/assert_eq.rs
@@ -46,3 +46,10 @@ fn assert_eq_custom() {
 
     assert_eq!(x, y, "custom panic message");
 }
+
+#[test]
+fn assert_eq_with_comparable_types() {
+	let s0: &'static str = "foo";
+	let s1: String = "foo".to_string();
+	assert_eq!(s0, s1);
+}

